IPs/RSKIP251.md

```diff
@@ -0,0 +1,71 @@
+# Support eth_getProof JSON RPC method
+
+|RSKIP          |251           |
+| :------------ |:-------------|
+|**Title**      |Support eth_getProof JSON RPC method |
+|**Created**    |30-06-2021 |
+|**Author**     |FJ |
+|**Purpose**    |Usa,Sec |
+|**Layer**      |Core |
+|**Complexity** |1 |
+|**Status**     |Draft |
+
+## Abstract
+
+RSK implements a Unitrie data structure to store all the relevant blockchain data, such as account states and storage values, this particular data structure enables an easy verification of each stored value, just by getting a merkle proof. Currently these proofs are only useful for blockchain nodes, but in order to allow verification of accounts outside the client, we need an additional function delivering us the required proof. 
+
+Combined with a stateRoot (from the blockheader) it enables offline verification of any account or storage-value. 
+
+## Motivation
+
+Offering these function through the RPC-Interface would enable applications to store and send proofs to devices which are not directly connected to the p2p-network and still are able to verify the data. This could be used for mobile applications, which are currently only using a remote client.
+
+Also it might be useful for L2 solutions.
+
+## Specification
+
+Returns the account and storage-values of the specified account including the Merkle-proof.
+
+Params
+- address (UNFORMATTED)
+- storage keys that we want to prove (if needed)  (UNFORMATTED[])
+- block number, block hash or an id (such as “earliest” or “latest”)  (UNFORMATTED)|(TAG)
+
+As response we’ll receive an Account object, composed by:
+
+- balance (QUANTITY)
+- codeHash (UNFORMATTED): SHA3 of the code associated to the specifica account (for EOA just retrieve SHA3(EMPTY_BYTE_ARRAY)) 
+- nonce (QUANTITY)
+- storageHash (UNFORMATTED): hash of the code of the account. For a EOA it will return "0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470" == SHA3(EMPTY_BYTE_ARRAY)
```

Comment:
hash of "code" -> hash of "value" in storage node/ cell (for the given the storage key)